hasher-matcher-actioner/hmalib/common/configs/tx_collab_config.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+DynamoDB backed CollaborationConfig objects.
+
+This may become a CollaborationConfigStoreBase, but since none of the pytx
+interfaces require it, keeping it as just another class for now.
+"""
+
+import typing as t
+import itertools
+
+from threatexchange.exchanges.collab_config import CollaborationConfigBase
+from threatexchange.exchanges.signal_exchange_api import SignalExchangeAPI
+
+from hmalib.common.config import HMAConfig, create_config
+
+
+def make_collab_config_subclass(
+    cls: t.Type[CollaborationConfigBase],
+) -> t.Type[HMAConfig]:
+    """
+    Generate an HMAConfig subclass on-the-fly.
+
+    Since supported APIs are dynamic, we need to allow any
+    CollaborationConfigBase to be stored as an HMAConfig object.
+    """
+    return type(f"_Dynamic{cls.__name__}_HMAConfig", (cls, HMAConfig), {})
+
+
+class CollabConfig(HMAConfig):
```

Comment:
Why `is-a` and not `has-a`?

Seems like an HMA config that just contained a collab config would still work, and require less magic. You also wouldn't need to worry about collisions then.

Also, dumb question, if this class isn't a dataclass, does that mess with the `__init__` form?

Reply:
Whoops, this module has stuff that I was experimenting with, but did not end up using. We are not using CollabConfig, it is safe to delete.

I'll push a new version. 

Essentially, for every CollabConfig subclass, we dynamically generate an HMAConfig type using the `make_collab_config_subclass` method above.